systemtest/gencorpora/gencorpora.go

```diff
@@ -0,0 +1,100 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gencorpora
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/elastic/apm-server/systemtest/loadgen"
+)
+
+// RunBlocking runs APM-Server and CatBulk server followed by
+// sending load to APM-Server and writing the generated corpus
+// to a file. RunBlocking exits on error or on successful
+// corpora generation.
+func RunBlocking(rootCtx context.Context) error {
```

Comment:
`RunBlocking` seems to imply that there is also a non-blocking function available, but I don't think that's the case.
Also, I would expect that `RunBlocking` blocks forever or until the context is cancelled, but that isn't the case here either.

 Would it make sense to just have a `Run()` method? The godoc comment above is informative enough where callers can make good decisions by just reading that.

Reply:
Agree with the logic, I would expect `RunBlocking` to run until I cancel the context. I have updated the function name to `Run`. Thanks for catching this.